datafusion/sql/src/expr/mod.rs

```diff
@@ -679,55 +661,6 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         )))
     }
 
-    fn parse_array_agg(
-        &self,
-        array_agg: ArrayAgg,
-        input_schema: &DFSchema,
-        planner_context: &mut PlannerContext,
-    ) -> Result<Expr> {
-        // Some dialects have special syntax for array_agg. DataFusion only supports it like a function.
```

Comment:
Sqlparser now handles this without a special case -- in https://github.com/sqlparser-rs/sqlparser-rs/pull/1247